### PR TITLE
Add tests for the circuit CLI

### PR DESCRIPTION
**Pull request by ltseeley**
Adds a number of tests for the `splinter circuit` CLI subcommands.
These tests verify the behavior of the subcommands, their arguments, and
their outputs.

All of these tests are disabled by default since they require special
setup to work properly, including modifying local files and starting the
Gameroom example application. These tests are considered a work in
progress.

Signed-off-by: Logan Seeley <seeley@bitwise.io>

To run the tests:

1. Startup the gameroom splinterd nodes in a terminal window:
```
$ docker-compose -f examples/gameroom/docker-compose.yaml up generate-key-registry splinterd-node-acme splinterd-node-bubba
```

2. In a new terminal window, copy Alice and Bob’s keys to the temp directory:
```
$ echo $(docker-compose -f examples/gameroom/docker-compose.yaml run generate-key-registry bash -c "cat /key_registry/alice.priv") | tee /tmp/alice.priv
$ echo $(docker-compose -f examples/gameroom/docker-compose.yaml run generate-key-registry bash -c "cat /key_registry/alice.pub") | tee /tmp/alice.pub
$ echo $(docker-compose -f examples/gameroom/docker-compose.yaml run generate-key-registry bash -c "cat /key_registry/bob.priv") | tee /tmp/bob.priv
$ echo $(docker-compose -f examples/gameroom/docker-compose.yaml run generate-key-registry bash -c "cat /key_registry/bob.pub") | tee /tmp/bob.pub
```

3. Run the tests from the CLI root directory:
```
$ cargo test --all-features -- --ignored --nocapture
```

~These tests require the changes in #663 to work properly~

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -20,13 +20,16 @@ extern crate diesel;
 mod action;
 mod error;
 mod store;
-
 #[cfg(feature = "circuit-template")]
 mod template;
+#[cfg(test)]
+mod tests;
```
[the remark this answers is not in the document]
Comment (ltseeley) [reply cli/src/main.rs:23]: Unfortunately that won't work; the logger is initialized per process, and the tests all run in one process. As far as I know, there isn't a way to reliably determine the order tests run in, so we can't know which `run` call will take place first. Also, there doesn't seem to be a way to check if a logger is already initialized with flexi_logger, so this is the only approach I could think of that would work.

File: cli/src/tests/circuit/mod.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        .skip_while(|item| item != &"Circuit:")
+        .skip(1)
+        .next()
+        .expect("Circuit ID not found in output")
+        .to_string()
+}
+
+/// Polls the splinter REST API at the given URL until all of the given circuit IDs show up in the
+/// node's list of proposals.
+fn wait_until_proposals_submitted(url: &str, circuit_ids: &[&str]) {
```
[the remark this answers is not in the document]
Comment (ltseeley) [reply cli/src/tests/circuit/mod.rs]: There is, it's provided by the `repeat_until_true` function. I added to the doc comments to indicate this timeout.

Ultimately, these functions will be replaced with an app auth handler so we don't have to poll for changes.

**Comment by ltseeley** [comment]
> The API-doc on the tests themselves could more clearly spell out what is being tested specifically.

Added more details to the doc comments to explain the test procedures

File: cli/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (vaporos) [reply cli/src/main.rs:23]: ok, "ignore_logger_init_error" would work though.

**Pull request merged**